console/console-init/ui/src/modules/iot-device/utils/TableFormatter.tsx

```diff
@@ -17,16 +17,18 @@ export const getTableCells = (row: IDevice) => {
         header: "id",
         title: (
           <span>
-            <Link to={"/"}>{row.id}</Link>
+            <Link to={`devices/${row.deviceId}/device-info`}>
+              {row.deviceId}
+            </Link>
           </span>
         )
       },
       {
         header: "type",
-        title: row.type && <span>{row.type}</span>
+        title: row.viaGateway ? <span>Using gateways</span> : <span>N/A</span>
```

Comment:
Add conditions for directly connected as well. If a device is having gateways in the jsonData then it is via gateway, and if device is having credentials then it directly connected and if device is `having both i.e gateways and credentials` OR `having nothing` then it is N/A